force-app/main/default/classes/STO_SurveyHelper.cls

```diff
@@ -1,20 +1,53 @@
 public without sharing class STO_SurveyHelper {
-    @AuraEnabled
-    public static Boolean canSendSurvey() {
-        return true;
+    /**
+     * @description Function to get InvitationLink for a Case
+     * @param Id caseId
+     * @return String InvitationLink
+     */
+    @AuraEnabled(cacheable=true)
+    public static String getSurveyLink(Id caseId) {
+        return [SELECT InvitationLink FROM SurveyInvitation WHERE Name = :caseId LIMIT 1]?.InvitationLink;
     }
 
-    @AuraEnabled
-    public static String getSurveyLink() {
-        return [
-            SELECT Id, InvitationLink
-            FROM SurveyInvitation
-            WHERE
-                OptionsAllowGuestUserResponse = TRUE
-                AND IsDefault = TRUE
-                AND Survey.DeveloperName = 'nks_service_survey'
-            LIMIT 1
-        ]
-        ?.InvitationLink;
+    /**
+     * @description Function to create survey link
+     * Calls function from Survey-repo
+     * @param List<Id> caseId
+     * @return List<String> survey link
+     */
+    @invocableMethod(label='Generate SurveyInvitationLink')
+    public static List<String> generateSurveyLink(List<Id> caseIds) {
+        List<String> surveyLinks = new List<String>();
+
+        try {
+            Id surveyId = [SELECT Id, DeveloperName FROM Survey WHERE DeveloperName = 'nks_service_survey' LIMIT 1]?.Id;
```

Comment:
Burde denne hardkodes? Kan vi ikke legge det til som et variabel i funksjonskallet?

Reply:
Godt poeng; problemet er at invocable funksjoner tar bare en parameter. Selve funksjonen som kalles fra Survey-repo tar to input.

Reply:
Du kan lage en egen invocable class og bruke den som input i Invocable funksjonen